osu.Game.Rulesets.Mania/Skinning/LegacyBodyPiece.cs

```diff
@@ -92,11 +102,25 @@ private void load(ISkinSource skin, IScrollingInfo scrollingInfo, DrawableHitObj
                 InternalChild = bodySprite;
 
             direction.BindTo(scrollingInfo.Direction);
-            direction.BindValueChanged(onDirectionChanged, true);
-
-            var holdNote = (DrawableHoldNote)drawableObject;
             isHitting.BindTo(holdNote.IsHitting);
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            direction.BindValueChanged(onDirectionChanged, true);
             isHitting.BindValueChanged(onIsHittingChanged, true);
+            missFadeTime.BindValueChanged(onMissFadeTimeChanged, true);
+
+            holdNote.ApplyCustomUpdateState += applyCustomUpdateState;
+            applyCustomUpdateState(holdNote, holdNote.State.Value);
+        }
+
+        private void applyCustomUpdateState(DrawableHitObject hitObject, ArmedState state)
+        {
+            if (state == ArmedState.Miss)
+                missFadeTime.Value ??= hitObject.StateUpdateTime;
```

Comment:
Shouldn't this be `HitStateUpdateTime`? Also what case does this code block handle - doesn't everything go through HoldNoteBroken?

Reply:
Yeah, it should be the hit variant, my mistake - however this *is* required in case the hold note head is never hit, as such a note doesn't really get broken. Open to suggestions how this could be done better.